lib/matplotlib/axis.py

```diff
@@ -1102,7 +1102,12 @@ def _update_ticks(self, renderer):
             tick.set_label1(label)
             tick.set_label2(label)
             if not mtransforms.interval_contains(interval_expanded, loc):
+                tick.set_visible(False)
+                tick.label1.set_visible(False)
```

Comment:
either also mark the gridline and ticks themselves as invisible, or add a comment about what's happening

Reply:
I have written in many places that we should be stricter and require the ticker to only return visible ticks (after all it has all the info needed to do so).  But I guess that's not happening any time soon and instead we just move the issues further down the stack :/

Reply:
Agenda item for today’s call!